### PR TITLE
Added /mobile route to show the QR 

### PR DESCRIPTION
**Pull request by shreya-mishra**
**Ticket** 
https://github.com/Real-Dev-Squad/website-my/issues/367

**Design for now** 
image

### CODE REVIEW

File: app/templates/index.hbs [hunk 1]
```diff
@@ -2,6 +2,19 @@
   <h1>Welcome to my site!</h1>
 </div>
 
+{{#if this.isDev}}
```
Comment (rohan09-raj) [inline app/templates/index.hbs:5]: Can you please verify if this check is working or not? I am trying to visit the same with `dev=true` flag but it isn't working.
image

File: app/templates/index.hbs [hunk 2]
```diff
@@ -2,6 +2,19 @@
...
+<div class='qrCodeContainer'> 
+<div class='qrSection'>
+  <QrCode
+    @text='http://example.com'
+    class='qrcode'
+    @darkColor='#00000'
```
Comment (rohan09-raj) [inline app/templates/index.hbs:11]: Can we make a separate component for this QR code feature?

**Review by rohan09-raj — CHANGES_REQUESTED** [review]

File: app/templates/index.hbs (hunk 2 above)
Comment (shreya-mishra) [reply app/templates/index.hbs:11]: sure

File: app/templates/index.hbs (hunk 1 above)
Comment (shreya-mishra) [reply app/templates/index.hbs:5]: resolved it in last commit

File: app/templates/index.hbs [hunk 3]
```diff
@@ -2,6 +2,19 @@
...
+  />
+  <p class='qrCodeText'>Check out our RDS mobile app</p>
```
[the remark this answers is not in the document]
Comment (shreya-mishra) [reply app/templates/index.hbs]: done

File: app/controllers/index.js [hunk 4]
```diff
@@ -10,7 +10,6 @@ const BASE_URL = ENV.BASE_API_URL;
 
 export default class IndexController extends Controller {
   @service toast;
-  @tracked status = this.model;
```
Comment (rohan09-raj) [inline app/controllers/index.js:13]: Why we have removed this, I think it's used below ?

File: app/styles/app.css [hunk 5]
```diff
@@ -47,6 +47,25 @@ body {
...
+  margin-top: 50px;
+  display: flex;
+  text-align: center;
+}
+
+.qr-code-text{
+  text-align: center;
+  margin-top:15px;
+}
+
```
Comment (rohan09-raj) [inline app/styles/app.css:68]: Can we please create a new CSS file for the new template?

File: app/templates/mobile.hbs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+{{page-title "Mobile"}} 
+<div class='qr-code-container'> 
+ <QrCode
+    data-test="qr-code"
+    @text= {{@model.userId}}
+    @darkColor='#00000'
+    class = 'qr-code'
+  />
+ </div>
+ <p class='qr-code-text'>Check out our RDS mobile app</p>
```
Comment (rohan09-raj) [inline app/templates/mobile.hbs:10]: Why this is being added at multiple places,
- We have component for the same we can simply call that
- It's in `index.hbs` as well, can we please remove it

**Review by rohan09-raj — CHANGES_REQUESTED** [review]

File: app/controllers/index.js (hunk 4 above)
Comment (shreya-mishra) [reply app/controllers/index.js:13]: check mobile.js for more context

File: app/routes/mobile.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+          '',
+          toastNotificationTimeoutOptions
+        );
+        return { status: USER_STATES.DNE, userId: userData.userId };
+      }
+    } catch (error) {
+      console.error(error.message);
+    }
+  };
+}
```
Comment (pallabez) [inline app/routes/mobile.js:44]: We should in future use ember data to get states.

**Review by pallabez — DISMISSED** [review]
LGTM

File: app/controllers/index.js (hunk 4 above)
Comment (rohan09-raj) [reply app/controllers/index.js:13]: I think this is different and was not added by you in `Index Controller`, this is being used for tracking the status of the user whether they are ooo, active , or idle

File: app/controllers/mobile.js [hunk 8]
```diff
@@ -0,0 +1,3 @@
+import MobileRoute from '../routes/mobile';
+
+export default class MobileController extends MobileRoute {}
```
Comment (rohan09-raj) [inline app/controllers/mobile.js:3]: Why controller is extending MobileRoute here? I think it should be `Controller`. Check other controller files for reference

File: app/routes/mobile.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+        // added setTimeout here because before new page opens user should be notified of error by toast
+        setTimeout(redirectAuth, 2000);
+      } else if (response.status === 404) {
+        this.toast.error(
+          `Your Status data doesn't exist yet. Please choose your status from the options below.`,
+          '',
+          toastNotificationTimeoutOptions
+        );
+        return { status: USER_STATES.DNE, userId: userData.userId };
+      }
```
Comment (rohan09-raj) [inline app/routes/mobile.js:39]: I don't think we need the `/users/status/self` call here. We can simply call `/users/self` to check the `Unauthorized` state (401) and `Success` (200) state and handle it accordingly (for reference line 24 to 31)
Rest all logic can be removed as of now as we don't need `status` data or userId for `/mobile`
Can be added if there is a need in future

File: app/styles/app.css (hunk 5 above)
Comment (rohan09-raj) [reply app/styles/app.css:68]: Can we please remove the duplicated CSS from here ? and the filename could be same as route name as per our general convention for component and template specific CSS.

For components - CSS file name should be same as component name
For templates - CSS file name should be same as template name

File: app/components/qrcode.hbs [hunk 10]
```diff
@@ -0,0 +1,10 @@
+
+<div class='qr-code-container'> 
+ <QrCode
+    data-test="qr-code"
+    @text= {{@text}}
+    @darkColor='#00000'
+    class = 'qr-code'
+  />
+ </div>
+ <p class='qr-code-text'>Check out our RDS mobile app</p>
```
Comment (rohan09-raj) [inline app/components/qrcode.hbs:10]: Is this used somewhere ? The code looks similar to `mobile.hbs` template

File: app/templates/mobile.hbs (hunk 6 above)
Comment (rohan09-raj) [reply app/templates/mobile.hbs:10]: Here, `Qrcode` component can be called which you have created and then line 2 to 10 can be removed

File: tests/integration/components/qrcode-test.js [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+import hbs from 'htmlbars-inline-precompile';
+import { setupRenderingTest } from 'ember-qunit';
+
+module('Integration | Component | qr-code', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('QR code component and text render on mobile route', async function (assert) {
+    this.set('userId', '12345');
+    await render(hbs`
+     <Qrcode @text = {{this.userId}} />
```
Comment (rohan09-raj) [inline tests/integration/components/qrcode-test.js:12]: We are testing here the component i.e. `qrcode.hbs` but what is the point of testing this component when we aren't using it ?
For reference, please check this comment

**Review by rohan09-raj — CHANGES_REQUESTED** [review]

File: tests/integration/components/qrcode-test.js (hunk 11 above)
Comment (shreya-mishra) [reply tests/integration/components/qrcode-test.js:12]: fixed

File: app/components/qrcode.hbs (hunk 10 above)
Comment (shreya-mishra) [reply app/components/qrcode.hbs]: removed

File: app/controllers/mobile.js (hunk 8 above)
Comment (shreya-mishra) [reply app/controllers/mobile.js]: replaced

File: app/controllers/index.js (hunk 4 above)
Comment (shreya-mishra) [reply app/controllers/index.js]: agreed

File: app/routes/mobile.js (hunk 9 above)
Comment (rohan09-raj) [reply app/routes/mobile.js:39]: You can grab that from /users/self call only

Comment (shreya-mishra) [reply app/routes/mobile.js]: done  in last commit .

**Review by rohan09-raj — APPROVED** [review]

**Pull request merged**